Propose branching strategy (#23)
buerokratt/Cross-functional-requirements: docs/technical/branching-strategy.md

# Branching Strategy

This document outlines the overall branching strategy for all components repositories in the Bürokratt system.

[Trunk Based Development](https://trunkbaseddevelopment.com/) model is used.

Feature branches should only be merge into main via **pull request** - **PR**

PRs can only be merged into main with **squash merge**

For branching prefexes are used:
- release/*
- feature/*
- hotfix/*

> Feature branch name should consist of prefex followed by issue number and issue title in lowercase, concatinated wiht hyphens.

Reviewer: ```suggestion
Feature branch name should consist of prefix followed by issue number and issue title in lowercase, concatenated with hyphens.
```

# Branching Strategy

This document outlines the overall branching strategy for all components repositories in the Bürokratt system.

[Trunk Based Development](https://trunkbaseddevelopment.com/) model is used.

# TL;DR
* Short living feature branches

> * Feature branch goes back to main through green light from CI pipeline and code review.

Reviewer 2: ```suggestion
* Feature branch goes back to `main` branch after it passes the CI pipeline and code review.
```

> * Feature branch goes back to `main` branch after it passes the CI pipeline and code review.

Reviewer: A few more things that might be worth mentioning:
- Feature branches should only be merge into `main` via PR (might be worth being explicit)
- PRs can only be merged into `main` with squash merge

@turnerrainer mentioned that he wanted to handle releases as branches so having some naming conventions might help manage this. It might be better to suggest that branch names should be prefixed with one the popular conventions:
- `release/*` (these can also be tags)
- `feature/*`
- `hotfix/*`

What do you think?

Reviewer 3: I didn´t follow it myself, will do it from now on.

Author: What about rollbacks? If there is a need for it should it be also done via PR? And will it be done through `hotfix` prefix?
